### DISCUSSION TITLE
Cypress TypeError: Cannot read properties of undefined (reading 'uid')

### DISCUSSION THREAD

**Discussion by ashokkumarg** [cypress-io/cypress#19649] [Cypress Cloud]
I have a Cypress Typescript with Cucumber project and all of a sudden it started to throw the below exception

`TypeError: Cannot read properties of undefined (reading 'uid')`

Some times it works when I change to a different workspace but again this error is reproducing when I open the Cypress runner itself and even when in executing in Jenkins too.

Cypress Runner: (v8.4.0)
![1](https://user-images.githubusercontent.com/7566850/148940706-be9dfd2d-b5be-4832-b32f-92b10c1d8762.png)

Jenkins Console: (v8.7.0)

> The function exported by the plugins file threw an error.
> 
> We invoked the function exported by `C:\Jenkins\workspace\ABC\cypress\plugins\index.js`, but it threw an error.
> 
>  TypeError: Cannot read properties of undefined (reading 'uid')
>     at Object.statSync (C:\Windows\SysWOW64\config\systemprofile\AppData\Local\Cypress\Cache\8.7.0\Cypress\resources\app\packages\server\node_modules\graceful-fs\polyfills.js:303:17)
>     at isDirectory (C:\Jenkins\workspace\ABC\ui\node_modules\resolve\lib\sync.js:22:23)
>     at loadNodeModulesSync (C:\Jenkins\workspace\ABC\ui\node_modules\resolve\lib\sync.js:191:17)
>     at Function.resolveSync [as sync] (C:\Jenkins\workspace\ABC\ui\node_modules\resolve\lib\sync.js:98:17)
>     at module.exports (C:\Jenkins\workspace\ABC\ui\cypress\plugins\index.js:15:25)
>     at C:\Windows\SysWOW64\config\systemprofile\AppData\Local\Cypress\Cache\8.7.0\Cypress\resources\app\packages\server\lib\plugins\child\run_plugins.js:90:12
>     at tryCatcher (C:\Windows\SysWOW64\config\systemprofile\AppData\Local\Cypress\Cache\8.7.0\Cypress\resources\app\packages\server\node_modules\bluebird\js\release\util.js:16:23)
>     at Function.Promise.attempt.Promise.try (C:\Windows\SysWOW64\config\systemprofile\AppData\Local\Cypress\Cache\8.7.0\Cypress\resources\app\packages\server\node_modules\bluebird\js\release\method.js:39:29)
>     at load (C:\Windows\SysWOW64\config\systemprofile\AppData\Local\Cypress\Cache\8.7.0\Cypress\resources\app\packages\server\lib\plugins\child\run_plugins.js:87:7)
>     at EventEmitter.<anonymous> (C:\Windows\SysWOW64\config\systemprofile\AppData\Local\Cypress\Cache\8.7.0\Cypress\resources\app\packages\server\lib\plugins\child\run_plugins.js:198:5)
>     at EventEmitter.emit (events.js:376:20)
>     at process.<anonymous> (C:\Windows\SysWOW64\config\systemprofile\AppData\Local\Cypress\Cache\8.7.0\Cypress\resources\app\packages\server\lib\plugins\util.js:19:22)
>     at process.emit (events.js:376:20)
>     at process.emit (C:\Windows\SysWOW64\config\systemprofile\AppData\Local\Cypress\Cache\8.7.0\Cypress\resources\app\packages\server\node_modules\source-map-support\source-map-support.js:495:21)
>     at emit (internal/child_process.js:910:12)
>     at processTicksAndRejections (internal/process/task_queues.js:83:21)
> error Command failed with exit code 1.

**Comment by aidanhyland** [comment]
Any update on this @ashokkumarg ?

**Reply by aidanhyland** [reply]
Tracked the issue here too. 

https://github.com/cypress-io/cypress/issues/19674

**Reply by ashokkumarg** [reply]
No @aidanhyland  Still the same issue exist

**Reply by san-slysz** [reply]
Also got this error after an upgrade. Also using Cucumber and Typescript.

**Comment by san-slysz** [comment]
@ashokkumarg doc in cucumber repo changed.
Changing my former code (identical to yours) with this one solved it for me.
I changed path.resolve('..') by path.resolve('.') because it suitted better my situation.

https://github.com/TheBrainFamily/cypress-cucumber-preprocessor#with-out-of-the-box-support

Hope it helps.

**Comment by michalbudzynski** [comment]
https://github.com/cypress-io/cypress/discussions/19539
